src/cartridges/int_adyen_SFRA/cartridge/client/default/js/applePayExpress.js

```diff
@@ -123,182 +122,177 @@ function callPaymentFromComponent(data, resolveApplePay, rejectApplePay) {
-                  amount: `${applePayButtonConfig.amount.value / 100}`,
-                },
-              };
-              resolve(finalPriceUpdate);
-            };
+    const applePayConfig = applePayPaymentMethod.configuration;
 
-            await callPaymentFromComponent(
-              { ...stateData, customer },
-              resolveApplePay,
-              reject,
-            );
-          } catch (error) {
-            reject(error);
-          }
-        },
-        onSubmit: () => {
-          // This handler is empty to prevent sending a second payment request
-          // We already do the payment in paymentFromComponent
-        },
-        onShippingMethodSelected: async (resolve, reject, event) => {
-          const { shippingMethod } = event;
-          const matchingShippingMethod =
-            shippingMethodsData.shippingMethods.find(
-              (sm) => sm.ID === shippingMethod.identifier,
-            );
-          const calculationResponse = await fetch(
-            `${window.calculateAmountUrl}?${new URLSearchParams({
-              shipmentUUID: matchingShippingMethod.shipmentUUID,
-              methodID: matchingShippingMethod.ID,
-            })}`,
-            {
-              method: 'POST',
+    const applePayButtonConfig = {
+      showPayButton: true,
+      configuration: applePayConfig,
+      amount: checkout.options.amount,
+      requiredShippingContactFields: ['postalAddress', 'email', 'phone'],
+      requiredBillingContactFields: ['postalAddress', 'phone'],
+      shippingMethods: shippingMethodsData.shippingMethods.map((sm) => ({
+        label: sm.displayName,
+        detail: sm.description,
+        identifier: sm.ID,
+        amount: `${sm.shippingCost.value}`,
+      })),
+      onAuthorized: async (resolve, reject, event) => {
+        try {
+          const customerData = event.payment.shippingContact;
+          const billingData = event.payment.billingContact;
+          const customer = formatCustomerObject(customerData, billingData);
+          const stateData = {
+            paymentMethod: {
+              type: APPLE_PAY,
+              applePayToken: event.payment.token.paymentData,
             },
-          );
-          if (calculationResponse.ok) {
-            const newCalculation = await calculationResponse.json();
-            applePayButtonConfig.amount = {
-              value: newCalculation.grandTotalAmount.value,
-              currency: newCalculation.grandTotalAmount.currency,
-            };
-            const applePayShippingMethodUpdate = {
+            paymentType: 'express',
+          };
+
+          const resolveApplePay = () => {
+            const finalPriceUpdate = {
               newTotal: {
                 type: 'final',
                 label: applePayConfig.merchantName,
-                amount: newCalculation.grandTotalAmount.value,
+                amount: `${applePayButtonConfig.amount.value / 100}`,
```

Comment:
If it is a foreign currency with 1 or 3 decimals, then this might not be correct. Should we have a check here?